### DISCUSSION TITLE
join on a subquery works with sqlalchemy 1.4.50 but not with 2.0.0 and up...

### DISCUSSION THREAD

**Discussion by konetsu** [sqlalchemy/sqlalchemy#10571] [Usage Questions] [answered]
trying to get the list of posts with the most amount of comments in the last 24 hours like this:
```
        recent_comments = Comment.query.filter(Comment.date  > datetime.now()-timedelta(hours = 24))
        hot_topics = db.session.query(Post).join(recent_comments).group_by(Post.id).order_by(func.count().desc()).limit(10).all()

```

this works just fine on the older versions, but if i update sqlalchemy i get this error:
`Don't know how to join to <sqlalchemy.sql.selectable.Select object at 0x0000020C64778ED0>. Please use the .select_from() method to establish an explicit left side, as well as providing an explicit ON clause if not present already to help resolve the ambiguity.`

how do i fix this? i'm using postgres btw.

**Comment by gordthompson** [comment, marked as answer]
Try explicitly declaring `recent_comments` as a subquery:

```python
    recent_comments = Comment.query.filter(
        Comment.date > datetime.now() - timedelta(hours=24)
    ).subquery()
```

**Reply by konetsu** [reply]
thanks. it did the trick :) weird how it works on older version but new one says "Don't know how to" lol.

**Reply by CaselIT** [reply]
On older version it likely raised a depreciation warning
